odbc/src/api/free_stmt_tests.rs

```diff
@@ -0,0 +1,137 @@
+mod unit {
+    use crate::{
+        handles::definitions::{
+            BoundColInfo, Connection, ConnectionState, Env, EnvState, MongoHandle, Statement,
+            StatementState,
+        },
+        map, SQLFreeStmt,
+    };
+    use bson::doc;
+    use definitions::{FreeStmtOption, Nullability, SqlReturn};
+    use mongo_odbc_core::json_schema::simplified::{Atomic, Schema};
+    use mongo_odbc_core::json_schema::BsonTypeName;
+    use mongo_odbc_core::{mock_query::MongoQuery, MongoColMetadata, MongoStatement};
+    use mongo_odbc_core::{Error, TypeMode};
+    use std::ptr::null_mut;
+
+    fn create_stmt_handle() -> *mut MongoHandle {
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        stmt
+    }
+
+    #[test]
+    fn test_free_stmt_invalid() {
+        let stmt = create_stmt_handle();
+        unsafe { assert_eq!(SqlReturn::ERROR, SQLFreeStmt(stmt as *mut _, 1)) }
+    }
+
+    #[test]
+    fn test_free_stmt_reset_params() {
+        let stmt = create_stmt_handle();
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLFreeStmt(stmt as *mut _, FreeStmtOption::SQL_RESET_PARAMS as i16)
+            )
+        }
+    }
+
+    #[test]
+    fn test_free_stmt_unbind() {
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
```

Comment:
Since I manipulate the `stmt` value in the `unsafe` block below, I am unable to use the helper function here (or in the next test). This is ok since these are just unit tests and this is only 4 lines of "repeated" code.